src/conventions/h-group/clue-interpretation/interpret-clue.js

```diff
@@ -250,9 +250,23 @@ export function interpret_clue(game, action) {
 	if (chop) {
 		focus_thoughts.chop_when_first_clued = true;
 
-		// A save is important if no one else could have given the save.
-		if (!old_focus_thoughts.saved && focus_thoughts.saved && (giver + 1) % state.numPlayers == target && !common.thinksLoaded(state, target, {assume: false}))
-			action.important = true;
+		// Check whether this is an urgent save.
+		if (!old_focus_thoughts.saved && focus_thoughts.saved && !common.thinksLoaded(state, target, {assume: false})) {
+			// If there is at least one non-finessed player between us and target, the save was not urgent.
+			let urgent = true;
+			let playerIndex = (giver + 1) % state.numPlayers;
+
+			while (playerIndex !== target) {
+				if (!state.hands[playerIndex].some(c => common.thoughts[c.order].finessed && state.isPlayable(c))) {
```

Comment:
There are a few intricacies that could be added here:
- If there is more than one finessed player, then an earlier player could play something that would make a finessed card in a later player's become playable. You'd need to figure out which finessed card each player would play (likely the one with the oldest `finesse_index`) and update a copy of the state/stacks.
- A player may have an unknown hidden finessed card that is playable, but the identity of their promised card is not yet playable. This player will not play on their turn, so they could clue.


This should be similar to [the logic that used to be used in clue_safe()](https://github.com/WillFlame14/hanabi-bot/blob/e0de01ae8658d0c858dfc01b03e927351e39d583/src/conventions/h-group/clue-finder/clue-safe.js#L28), if you want to copy it.

Also, should the comment should read "between the giver and the target"?

Reply:
Thanks! I've added something similar, though rethinking the logic for this particular case. In particular, I want to use the common thoughts so that everyone should come to the same conclusion about whether the clue is allowed or not.